src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs

```diff
@@ -341,7 +341,7 @@ internal static DataContract GetRevisedItemContract(DataContract oldItemContract
                 oldItemContract.UnderlyingType.GetTypeInfo().IsGenericType &&
                 (oldItemContract.UnderlyingType.GetGenericTypeDefinition() == Globals.TypeOfKeyValue))
             {
-                return ClassDataContract.CreateClassDataContractForKeyValue(oldItemContract.UnderlyingType, oldItemContract.Namespace, new string[] { JsonGlobals.KeyString, JsonGlobals.ValueString });
+                return DataContract.GetDataContract(oldItemContract.UnderlyingType);
```

Comment:
Yes indeed it was a fix we made on Desktop a few years ago but didn't get ported to CoreClr.